app/src/main/java/com/simply/birthdayapp/main/addEvent/data/repository/BirthdayRepositoryImpl.kt

```diff
@@ -0,0 +1,68 @@
+package com.simply.birthdayapp.main.addEvent.data.repository
+
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.api.Optional
+import com.simply.CreateBirthdayMutation
+import com.simply.birthdayapp.commondomain.model.Birthday
+import com.simply.birthdayapp.core.result.Result
+import com.simply.birthdayapp.main.addEvent.data.mapper.toBirthday
+import com.simply.birthdayapp.main.addEvent.data.mapper.toCreateBirthdayInputData
+import com.simply.birthdayapp.main.addEvent.data.model.BirthdayDataModel
+import com.simply.birthdayapp.main.addEvent.domain.model.CreateBirthdayInputDomain
+import com.simply.birthdayapp.main.addEvent.domain.repository.BirthDayRepository
+import com.simply.type.CreateBirthdayInput
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+
+class BirthdayRepositoryImpl(
+    private val apolloClient: ApolloClient
+) : BirthDayRepository {
+
+    override fun createBirthday(input: CreateBirthdayInputDomain): Flow<Result<Birthday>> = flow {
+        emit(
+            Result.Loading(
+                Birthday()
+            )
+        )
+        val birthdayInput = CreateBirthdayMutation(
+            CreateBirthdayInput(
+                date = input.toCreateBirthdayInputData().date,
```

Comment:
please create  val inputModel = input.toCreateBirthdayInputData(), and use this after that